src/reporter.ts

```diff
@@ -0,0 +1,164 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import type {
+  JUnitTestSuite,
+  JUnitTestCase,
+  Failure,
+  Error,
+  Property,
+  ConfigOption,
+} from './type';
+import { create } from 'xmlbuilder2';
+
+export class TestSuite implements JUnitTestSuite {
+  name: string;
+  tests: number;
+  failures: number;
+  errors: number;
+  time: number;
+  timestamp: string;
+  file: string;
+  testsuites: JUnitTestSuite[] | undefined;
```

Comment:
In my example repo I removed the `undefined` case, because it's simply not necessary. You are already initializing it [now at all times](https://github.com/saucelabs/cypress-junit-plugin/pull/1/files#diff-f5d7562fe14b5c4c08e44380f423efbc7bdf7ccdd0e799917503eec5334ed633R34).
```suggestion
  testSuites: JUnitTestSuite[];
```

PS: Don't forget to change the interface too.
PPS: Should be `testSuites` here and not `testsuites`, which is then finally consistent with `testCases` (see my code branch).